uqcsbot/dominos_coupons.py

```diff
@@ -139,27 +157,88 @@ def _get_coupons(n: int, ignore_expiry: bool, keywords: List[str]) -> List[Coupo
             for coupon in coupons
             if any(coupon.keyword_matches(keyword) for keyword in keywords)
         ]
-    return coupons[:n]
 
+    if not source:
```

Comment:
This is a great idea. Given the coupons are never ordered, the coupons should be shuffled regardless of `source`.